tests/integration/tests/test_util/util.py

```diff
@@ -212,12 +213,12 @@ def wait_until_k8s_ready(
 
 def wait_for_dns(instance: harness.Instance):
     LOG.info("Waiting for DNS to be ready")
-    instance.exec(["k8s", "x-wait-for", "dns"])
+    instance.exec(["k8s", "x-wait-for", "dns", "--timeout", "20m"])
```

Comment:
Can we make this less than 20m? I don't think it looks too good on us to have long waits like this. (Just to be clear, this is not your fault).